### DISCUSSION TITLE
Is there a plan?

### DISCUSSION THREAD

**Discussion by jonfriesen** [drgomesp/midgarts#26] [General]
Hello, I'm curious if there is a plan with this project? While exploring the codebase I've noticed a few different directions (eg. sdl2, vulkan, rust, and a 2d branch).

It feels a bit like this repository is more of a playground to learn rendering concepts using the RO assets. Which is totally cool.

I love the idea of Go based tooling that assembles a RO like environment. Not necessarily an entire client, but something that a some innovative indie game dev can use to build the next RO.

**Comment by drgomesp** [comment]
Hey @jonfriesen! Thanks for your interest in the project. I really appreciate your feedback.

Indeed the branches you noticed on the repository are mostly experimental, especially the rust one. One of my goals with the project was to actually keep using Go and that's not going to change. However, I did face some challenges when dealing with graphics libraries and it really feels like the Rust ecosystem is way more rich in terms of options and ways of doing things. Nonetheless, Go will still be used and I plan on writing everything with OpenGL anyways, so that shouldn't be a problem.

> I love the idea of Go based tooling that assembles a RO like environment. Not necessarily an entire client, but something that a some innovative indie game dev can use to build the next RO.

Please tell me more, I'd love to understand more about this idea. I have a lot of ideas for the project. My vision really is to create an ecosystem of small projects that solve different problems both in the client and the server realms for playing RO these days. I wanted to create a sort of integrated ecosystem for servers in which players could potentially jump from server to server and take their characters with them. Not to mention lots of other things in that same direction.

In any case, the project is currently slowed down mostly because of my personal life which has taken much of my time for the past couple of years. I plan on getting back with full force to it early this year, and the challenge I need to tackle now to get to a decent state is actually to render maps, which is way more complicated than 2d sprites.

**Reply by jonfriesen** [reply]
Awesome! Glad to hear you're planning to keep going in Go.

I'm a huge fan of the RO art style, game play mechanics, and world in general. I think it'd be really cool if this project could create all of the different components without coupling to the network client. This would allow the creation of other games using the RO assets. Eg. someone could create a single player RO with a linear story, or a world exploration game.

All that said, I totally understand personal life getting in the way. I'm in a similar situation and plan to dabble in contributing here and there but probably won't be able to be a signifcant contributor right now. I also know absolutely nothing about opengl/sdl2 so there's lot of learning to do 😅 . 

What you've built already is super impressive! Nice work and thank you for doing it in the open 🙂

**Comment by drgomesp** [comment]
And by the way, I have setup a Discord server in case people want to discuss things in more detail: https://discord.gg/tP2k7zNayW
